browser/resources/settings/email_aliases_page/tests/email_aliases.test.tsx

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import { render, screen, fireEvent, waitFor, act } from '@testing-library/react'
+import { ManagePage } from '../email_aliases'
+import { localeRegex, clickLeoButton } from './test_utils'
+import {
+  EmailAliasesServiceInterface,
+  EmailAliasesServiceObserverInterface
+} from 'gen/brave/components/email_aliases/email_aliases.mojom.m'
+
+jest.mock('$web-common/locale', () => ({
+  getLocale: (key: string) => {
+    return key
+  },
+  formatMessage: (key: string, params: Record<string, string>) => {
+    return key
+  },
+  formatLocale: (key: string, params: Record<string, string>) => {
+    return key
+  }
+}))
+
+// Mock the email aliases service
+class MockEmailAliasesServiceInterface extends EmailAliasesServiceInterface {
+  private observerInit?:
+    (observer: EmailAliasesServiceObserverInterface) => void
+  constructor(observerInit?:
+    (observer: EmailAliasesServiceObserverInterface) => void) {
+    super()
+    this.observerInit = observerInit
+  }
+
+  addObserver(observer: EmailAliasesServiceObserverInterface) {
+    this.observerInit?.(observer)
+  }
+  removeObserver(observer: EmailAliasesServiceObserverInterface) {}
+
+  generateAlias = jest.fn()
+  updateAlias = jest.fn()
+  deleteAlias = jest.fn()
+  requestPrimaryEmailVerification = jest.fn()
+  cancelPrimaryEmailVerification = jest.fn()
+  logout = jest.fn()
+  showSettingsPage = jest.fn()
+}
+
+const createBindObserver =
+  (emailAliasesService: MockEmailAliasesServiceInterface) =>
+  (observer: EmailAliasesServiceObserverInterface) => {
+    emailAliasesService.addObserver(observer)
+    return () => {
+      emailAliasesService.removeObserver(observer)
+    }
+  }
+
+describe('ManagePage', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('shows loading state initially', async () => {
+    const mockEmailAliasesService = new MockEmailAliasesServiceInterface()
+    await act(async () => {
+      render(<ManagePage
+               emailAliasesService={mockEmailAliasesService}
+               bindObserver={createBindObserver(mockEmailAliasesService)}
+      />)
+    })
+    expect(document.querySelector('leo-progressring'))
+      .toBeInTheDocument()
+    expect(screen.getByText(localeRegex(
+      'emailAliasesConnectingToBraveAccount'))).toBeInTheDocument()
+  })
+
+  it('shows sign up form when no email is available', async () => {
+    const mockEmailAliasesService =
+      new MockEmailAliasesServiceInterface((observer) => {
+      observer.onLoggedOut()
```

Comment:
this only gets triggered when the user logs out, right? So opening a new page this would never fire? I guess this is related to my other questions about getting data for a fresh load of a page

Reply:
I realized all these observer methods are a bit confusing and don't show the overall structure. So I introduced an `AuthenticationStatus` state machine in mojom, which is part of `AuthState` and it is monitored by the single observer method `OnAuthStateChanged'.

This method is called by addObserver to make sure that the UI is up to date on a fresh load of the page. And of course whenever the state changes, it will be called again.